portfolio/src/main/java/com/google/sps/servlets/ShawarmaServlet.java

```diff
@@ -0,0 +1,41 @@
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonIOException;
+import com.google.gson.JsonSyntaxException;
+import com.google.sps.data.ShawarmaPlace;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.Reader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+/**
+ * Servlet that returns a list of shawarma places.
+ */
+@WebServlet("/shawarma")
+public class ShawarmaServlet extends HttpServlet {
+    private final String SHAWARMA_FILE = "shawarma.json";
+    private final Path PATH_SHAWARMA_FILE = Paths.get("WEB-INF/classes", SHAWARMA_FILE).toAbsolutePath();
```

Comment:
nit: `Paths.get("WEB-INF", "classes", SHAWARMA_FILE)` to make it platform independent. It shouldn't be a problem since you're using app engine, but it's a good practice not to use "/" to separate file paths in strings